### PR TITLE
Cardinality properties

### PR DESCRIPTION
**Pull request by hasumedic**
This PR introduces the ability to set a couple of properties to the "Cardinality" aggregation as per ES documentation:

https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-cardinality-aggregation.html

The 2 options are:
* precision_threshold (integer)
* rehash (boolean)

Both setters control that the parameter given is of the right type.

### CODE REVIEW

File: test/lib/Elastica/Test/Aggregation/CardinalityTest.php [hunk 1]
```diff
@@ -33,4 +33,92 @@ public function testCardinalityAggregation()
...
+     * @expectedException \InvalidArgumentException
+     * @param mixed $rehash
+     */
+    public function testInvalidRehash($rehash)
+    {
+        $agg = new Cardinality('rehash');
+        $agg->setRehash($rehash);
+    }
+
+    public function invalidRehashProvider()
```
Comment (ruflin) [inline test/lib/Elastica/Test/Aggregation/CardinalityTest.php]: I would suggest to make the 4 "provider" functions protected as they don't need to be public.

**Comment by ruflin** [comment]
Looks good, see comment. Can you update the CHANGELOG.md file?

File: test/lib/Elastica/Test/Aggregation/CardinalityTest.php (hunk 1 above)
Comment (hasumedic) [inline test/lib/Elastica/Test/Aggregation/CardinalityTest.php]: Thanks for the quick review @ruflin! The annotated data providers need to be public so they can be called using reflection by PHPUnit.
https://phpunit.de/manual/4.6/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers

Comment (ruflin) [inline test/lib/Elastica/Test/Aggregation/CardinalityTest.php]: Make sense.

**Comment by ruflin** [comment]
Merged. Thx.

**Pull request merged**